doc/developing_in_windows_msys.md

```diff
@@ -0,0 +1,140 @@
+
+    pacman -Syu
+
+Now restart the MSYS environment and run the following command to complete updates:
+
+    packman -Su
+
+## The MSYS Environment
+
+It's important to note that **all updates and package installation must be done within the MSYS environment**.  After making any package changes, `exit` all MSYS2 terminals and restart the desired build-environment.  This reminder is repeated multiple times throughout this guide.
+
+The other environments are layers on top of MSYS.  If packages are installed within them, the environment structure will be disrupted and compilation will likely fail.  If this happens, it may be necessary to reinstall MSYS.
+
+1. Rename the `msys64` folder to `msys64.bak`.
+
+2. Use the installer to reinstall MSYS2 to `msys64`.
+
+3. Copy packages from `msys64.bak/var/cache/pacman/pkg/` to the new installation to save download time and bandwidth.
+
+4. Use `pacman` from within the MSYS environment to install and update packages.
+
+5. After successfully building a project, it is safe to delete `msys64.bak`
+
+## The MING64 Environment
+
+Start by setting up the MING64 environment.  The following commands should be run within the MSYS environment.  `pacman -S` is used to install packages.  The `--needed` argument prevents packages from being reinstalled.
+
+    pacman -S --needed base-devel mingw-w64-x86_64-toolchain
+    pacman -S git mingw-w64-x86_64-cmake mingw-w64-x86_64-ninja \
+        mingw-w64-x86_64-gtest mingw-w64-x86_64-giflib \
+        mingw-w64-x86_64-libpng mingw-w64-x86_64-libjpeg-turbo 
+
+If any packages are found to be missing later, you may search for them with:
+
+    pacman -Ss [search-term]
+
+When searching for packages, results are in the form: `[group]/package-name`.  For example:
+
+    mingw32/mingw-w64-i686-cmake
+    mingw64/mingw-w64-x86_64-cmake
+    ucrt64/mingw-w64-ucrt-x86_64-cmake
+    clang64/mingw-w64-clang-x86_64-cmake
+    msys/cmake
+
+For tools that need to be aware of the compiler to function, install the package that corresponds with the specific build-environment you plan to use.  For `cmake`, install the `mingw64` version.  The generic `msys/cmake` will not function correctly because it will not find the compiler.  For other tools, the generic `msys` version is adequate, like `msys/git`.
+
+To remove packages, use:
+
+    pacman -Rsc [package-name]
+
+## Other Environments
+
+MSYS2 provides multiple development [environments](https://www.msys2.org/docs/environments/), which are layers on top of the MSYS environment.  They target slightly different platforms, runtime libraries, and compiler toolchains.  Remember to always make package changes from within the MSYS environment.  Then `exit` all MSYS2 environments and restart the desired build environment.
+
+* **MINGW32:**  To develop for 32-bit Windows, replace `x86_64` with `i686` in the package names. 
```

Comment:
Do you mean Mingw-w64?

Reply:
I was using the 64-bit version of MSYS2.  I don't know if the package names are different on the 32-bit version.  I'll rewrite to clarify.

Reply:
Sorry, you are right, the package group is indeed called mingw32/mingw64. I think I may have been thrown off by the upper case.

Reply:
I refer to the build environment in uppercase (which link to click in the Start Menu).  The group names look like the build environment names, but in lower case.  Then there's the prefix and rest of the package name.  I'll see if I can add an explanation to clarify, since confusing the differences contributed to difficulties building libjxl on MSYS2.